plugins/gopacket/run.go

```diff
@@ -0,0 +1,242 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/signal"
+	"path"
+	"plugin"
+	"sort"
+	"strconv"
+	"syscall"
+
+	logutils "github.com/Portshift/go-utils/log"
+	log "github.com/sirupsen/logrus"
+	"github.com/up9inc/mizu/shared/logger"
+	"github.com/up9inc/mizu/tap"
+	"github.com/up9inc/mizu/tap/api"
+	"github.com/urfave/cli"
+
+	"github.com/apiclarity/apiclarity/plugins/api/client/client"
+	"github.com/apiclarity/apiclarity/plugins/api/client/client/operations"
+	"github.com/apiclarity/apiclarity/plugins/api/client/models"
+	"github.com/apiclarity/apiclarity/plugins/common"
+	"github.com/apiclarity/apiclarity/plugins/gopacket/config"
+	"github.com/apiclarity/apiclarity/plugins/gopacket/monitor"
+)
+
+type Agent struct {
+	podMonitor *monitor.PodMonitor
+	apiClient  *client.APIClarityPluginsTelemetriesAPI
+}
+
+func run(c *cli.Context) {
+	logutils.InitLogs(c, os.Stdout)
+
+	runConfig := config.LoadConfig()
+	log.Infof("Loaded config: %+v", runConfig)
+
+	// load http plugin
+	extensions, err := loadExtensions()
+	if err != nil {
+		log.Errorf("Failed to load extensions: %v", err)
+		return
+	}
+	opts := &tap.TapOpts{
+		HostMode: true,
+	}
+
+	agent := &Agent{
+		apiClient: common.NewAPIClient(runConfig.UpstreamAddress),
+	}
+
+	// set mizu logger
+	logger.InitLoggerStderrOnly(runConfig.MizuLogLevel)
+
+	podMonitor, err := monitor.NewPodMonitor(runConfig.NamespaceToTap)
+	if err != nil {
+		log.Errorf("Failed to create pod monitor: %v", err)
+		return
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	agent.podMonitor = podMonitor
+	go podMonitor.Start(ctx)
+
+	outputItems := make(chan *api.OutputChannelItem)
+	options := &api.TrafficFilteringOptions{}
+	tap.StartPassiveTapper(opts, outputItems, extensions, options)
+
+	go agent.startReadOutputItems(ctx, outputItems)
+
+	// Wait for deactivation
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+
+	s := <-sig
+	log.Warningf("Received a termination signal: %v", s)
+	cancel()
+}
+
+func (a *Agent) startReadOutputItems(ctx context.Context, outputItems chan *api.OutputChannelItem) {
+	log.Info("Starting to read output items")
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case item := <-outputItems:
+			go func() {
+				log.Infof("Output item received: connection info: %+v", item.ConnectionInfo)
+				serverNamespace := a.podMonitor.GetPodNamespaceByIP(item.ConnectionInfo.ServerIP)
+				if a.shouldIgnoreTelemetry(item, serverNamespace) {
+					return
+				}
+				telemetry, err := a.createTelemetry(item, serverNamespace)
```

Comment:
as discussed, `serverNamespace` should be deleted from `createTelemetry`.
`serverNamespace := a.podMonitor.GetPodNamespaceByIP(item.ConnectionInfo.ServerIP)` need to go into `a.shouldIgnoreTelemetry`.